test/common/common.go

```diff
@@ -0,0 +1,446 @@
+//go:build e2e
+// +build e2e
+
+package common
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/miekg/dns"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	googleDNSServer            = "8.8.8.8"
+	e2eSeparateOperandNsEnvVar = "E2E_SEPARATE_OPERAND_NAMESPACE"
+	operandNamespace           = "external-dns"
+	rbacRsrcName               = "external-dns-operator"
+	operatorServiceAccount     = "external-dns-operator"
+	OperatorNamespace          = "external-dns-operator"
+	dnsProviderEnvVar          = "DNS_PROVIDER"
+	e2eSkipDNSProvidersEnvVar  = "E2E_SKIP_DNS_PROVIDERS"
+	DnsPollingInterval         = 15 * time.Second
+	DnsPollingTimeout          = 3 * time.Minute
+)
+
+func DefaultService(name, namespace string) *corev1.Service {
+	return testService(name, namespace, corev1.ServiceTypeLoadBalancer)
+}
+
+func testService(name, namespace string, svcType corev1.ServiceType) *corev1.Service {
```

Comment:
So I just copied and pasted the functions from the original `util.go`. This indirection was introduced by this commit: https://github.com/openshift/external-dns-operator/commit/f52e31052b4ec31164239b058dbb880320aea2dd, but I don't think it was ever used.

It looks like `clusterIPService` got removed in https://github.com/openshift/external-dns-operator/commit/c5de001aa96cace10ce9ecc36c20a4eeb415d4f4

I can remove the indirection now since I've touched it.